docs/src/ext_dd.md

```diff
@@ -1 +1,85 @@
-# Extending DimensionalData
\ No newline at end of file
+# Extending DimensionalData
+
+Nearly everything in DimensionalData.jl is designed to be extensible.
+
+- `AbstractDimArray` are easily extended to custom array types. `Raster` or
+  `YAXArray` are examples from other packages.
+- `AbstractDimStack` are easily extended to custom mixed array dataset.
+    `RasterStack` or `ArViZ.Dataset` are examples.
+- `LookupArray` can have new types added, e.g. to `AbstractSampled` or
+  `AbstractCategorical`. `Rasters.Projected` is a lookup that knows
+  its coordinate reference system, but otherwise behaves as a regular
+  `Sampled` lookup.
+
+`dims` and `rebuild` are the key interface methods in most of these cases.
+
+## `dims`
+
+Objects extending DimensionalData.jl that have dimensions must return 
+a `Tuple` of constructed `Dimension`s from `dims(obj)`. 
+
+### `Dimension` axes
+
+Dimensions return from `dims` should hold a `LookupArray` or in some cases 
+just an `AbstractArray` (like wiht `DimIndices`). When attached to 
+mullti-dimensional objects, lookups must be the _same length_ as the axis 
+of the array it represents, and `eachindex(A, i)` and `eachindex(dim)` must 
+return the same values. 
+
+This means that if the array has OffsetArrays.jl axes, the array the dimension 
+wraps must also have OffsetArrays.jl axes.
+
+### `dims` keywords
+
+To any `dims` keyword argument that only marks the dimension name,
+objects should accept any `Dimension`, `Type{<:Dimension}`, `Symbol`,
+`Val{:Symbol}`, `Val{<:Type{<:Dimension}}` or regular `Integer`. 
+This is easier than it sounds, calling `DD.dims(objs, dims)` will
+return the matching dimension and `DD.dimnum(obj, dims)` will return
+the matching `Int` for any of these inputs as long as `dims(obj)` is
+implemented.
+
+
+## `rebuild`
+
+Rebuild methods are used to rebuild immutable objects with new field values,
+in a way that is more flexible and extensible than just using ConstructionBase.jl
+reconstruction. Developers can choose to ignore some of the fields passed
+by `rebuild`.
+
+The function signature is always one of:
+
+```julia
+rebuild(obj, args...)
+rebuild(obj; kw...)
+```
+
+`rebuild` has keyword versions automatically generated for all objects
+using [ConstructionBase.jl](https://github.com/JuliaObjects/ConstructionBase.jl). 
+
+These will work without further work as long as your object has the fields 
+used by DimensionalData.jl objects. For example, `AbstractDimArray` will 
+receive these keywords in `rebuild`: `data`, `dims`, `refdims`, `name`, `metadata`. 
+
+If your `AbstractDimArray` does not have all these fields, you must implement
```

Comment:
Could I also fake these fields by overloading `getproperty(MyDimArray, :dims)? Or would this be a bad idea?